### PR TITLE
CDAP-3886 fix how rowkey is constructued for writing/deleting/listing feed entries

### PR DESCRIPTION
**Pull request by gsps1**
JIRA : https://issues.cask.co/browse/CDAP-3886

### CODE REVIEW

**Comment by gsps1** [comment]
build : http://builds.cask.co/browse/CDAP-DUT2916-1

**Comment by poornachandra** [comment]
@gsps1 Please add a test case for this fix.

**Comment by gsps1** [comment]
@poornachandra, have added a test case, please take a look

**Comment by gsps1** [comment]
http://builds.cask.co/browse/CDAP-DUT2916-3

File: cdap-notifications/src/test/java/co/cask/cdap/notifications/NotificationTest.java [hunk 1]
```diff
@@ -169,6 +167,38 @@ public void received(String notification, NotificationContext notificationContex
...
+    Assert.assertTrue(checkFeedList(ImmutableSet.of(FEED1, FEED2), feedManager.listFeeds(namespace)));
+
+    // clear the feeds
+    feedManager.deleteFeed(FEED1);
+    feedManager.deleteFeed(FEED2);
+    namespaceClient.delete(namespace);
+    Assert.assertEquals(0, feedManager.listFeeds(namespace).size());
+  }
+
+  private boolean checkFeedList(ImmutableSet<Id.NotificationFeed> expected, List<Id.NotificationFeed> actual) {
```
Comment (anwar6953) [inline cdap-notifications/src/test/java/co/cask/cdap/notifications/NotificationTest.java]: I think you should also assert that their size is equal.
Else the following assertion will pass, when it shouldn't:
`checkFeedList(  Set of size 5, List of size 0 );`

**Comment by anwar6953** [comment]
Besides one comment in test case, LGTM.

**Comment by gsps1** [comment]
have addressed the comment and rebased the commits, merging.

**Pull request merged**